File: fsharp/leap/Leap.fs
module Leap

let leapYear (year: int): bool =
    match year % 400 = 0, year % 100 = 0, year % 4 = 0 with
    | (true, _, _) -> true
    | (_, true, true) -> false
    | (_ , _, true) -> true
    | _ -> false

File: scala/leap/src/main/scala/Leap.scala
object Leap {
  def leapYear(year: Int): Boolean =
    (year % 4 == 0, year % 100 == 0, year % 400 == 0) match {
      case (_, _, true) => true
      case (_, true, _) => false
      case (true, _, _) => true
      case (_) => false
    }
}
